### DISCUSSION TITLE
Property 'session' does not exist on type 'IncomingMessage'.ts(2339) any

### DISCUSSION THREAD

**Discussion by Anand-Keshavan** [socketio/socket.io#5173] [Q&A]
I get the above error when I try to use the express-session example ( typescript)  in the documentation. Here is the code:

`declare module "express-session" {
    interface SessionData {
        
        count: number;
    }
}

const sessionMiddleware = session({
    secret: "losgatos",
    resave: true,
    saveUninitialized: true,
});

const app = express();
app.use(sessionMiddleware);
const httpServer = createServer(app);

const io = new Server(httpServer, { /* options */ });
io.engine.use(sessionMiddleware);

o.on("connection", (socket) => {
    const sessionId =  socket.request.session.id.  // <---- error here
... 

`
package versions:

`"@types/express": "^4.17.21",
    "@types/express-session": "^1.18.0",
    "express": "^4.19.2",
    "express-session": "^1.18.0",
    "socket.io": "^4.7.5",`

**Comment by darrachequesne** [comment]
Hi! You need to cast the `socket.request` as an Express `Request` object:

```ts
import { type Request } from "express";

io.on("connection", (socket) => {
  const req = socket.request as Request;

  // ...
});
```

Reference: https://socket.io/how-to/use-with-express-session
